Use platform/generator agnostic CMake commands (#2539)
raspberrypi/pico-sdk: README.md

1. Setup a CMake build directory.
      For example, if not using an IDE:
      ```
      $ cmake -S . -B build
      ```

>       > The cmake -S flag indicates the source directory, and -B tells cmake where to put the output files.

Reviewer: ```suggestion
      > The cmake -S flag indicates the source directory, and the -B flag tells cmake the name of the output-directory to create. This doesn't have to be named "build", you can call it whatever you want.
```

1. Setup a CMake build directory.
      For example, if not using an IDE:
      ```

>       $ cmake -S . -B build

Reviewer: IMHO we should either explain what these short command-line switches mean...

Author: Something like adding
> `cmake -S <source_dir> -B <build_dir>`

below the command?

Reviewer: Yeah, or just something like
> The cmake `-S` flag indicates the source directory, and `-B` tells cmake where to put the output files.

? :shrug:

1. Make your target from the build directory you created.
      ```sh

>       $ cmake --build build --target hello_world

Reviewer: ...or we should be consistent and just use the long command-line switches? (which are "more self-documenting" than the short options)

1. Make your target from the build directory you created.
      ```sh

>       $ cmake --build build --target hello_world

Author: I was wondering if it'd be cleaner to use `./build` or `build/` or something, to make it clearer it's a directory?

>       $ cmake --build build --target hello_world

Reviewer: Maybe a bit wordy, but I've suggested one possible way that we could explain this more clearly.

      ```

> 

Author: Probably as an inline quote, like the other similar comment above
```suggestion
> The directory-name supplied to the `--build` flag needs to match the directory-name that was passed to the `-B` flag in the earlier cmake command.

```

1. Make your target from the build directory you created.
      ```sh
      $ cmake --build build --target hello_world
      ```

> > The directory-name supplied to the `--build` flag needs to match the directory-name that was passed to the `-B` flag in the earlier cmake command.

Author: ```suggestion
      > The directory-name supplied to the `--build` flag needs to match the directory-name that was passed to the `-B` flag in the earlier cmake command.
```
